### PR TITLE
Fixing the dataloader seed for accuracy experiments

### PR DESCRIPTION
**Pull request by dgourab-aws**

### CODE REVIEW

File: axlearn/common/input_lm.py [hunk 1]
```diff
@@ -21,6 +21,9 @@
 # Value of "target_labels" which will be ignored in seq2seq processing.
 SEQ2SEQ_IGNORE_TARGET_LABEL = -1
 
+seed = os.environ.get("DATA_SEED")
+seed = int(seed) if seed is not None else None
+tf.random.set_seed(seed)
```
Comment (satishpasumarthi) [inline axlearn/common/input_lm.py:26]: would this line work if seed is `None` ?

Comment (dgourab-aws) [reply axlearn/common/input_lm.py:26]: Yes it will work as it's currently working, without any set seed.

**Comment by patrick-toulme** [comment]
Can you rebase and squash these commits?

File: axlearn/common/input_lm.py (hunk 1 above)
Comment (patrick-toulme) [inline axlearn/common/input_lm.py:26]: Axlearn already sets a seed?

**Review by patrick-toulme — CHANGES_REQUESTED** [review]

File: axlearn/common/input_lm.py (hunk 1 above)
Comment (dgourab-aws) [reply axlearn/common/input_lm.py:26]: I haven't seen Axlearn setting a seed anywhere, the input orders are random for every run. This PR is to fix that, we need to set the DATA_SEED as an environment variable in our launch script.

Comment (patrick-toulme) [reply axlearn/common/input_lm.py:26]: https://github.com/apple/axlearn/blob/a15a3bcbb976c14db157a8958df368a48c614c1f/axlearn/common/input_tf_data.py#L383

Comment (dgourab-aws) [reply axlearn/common/input_lm.py:26]: This seed is to sample from multiple data sources, I dont think this is to fix the order of samples in 1 data source.

**Pull request merged**